pytentiostat/config_reader.py

```diff
@@ -37,10 +36,121 @@ def parse_config_file(configlocation=None):
             with open(os.path.join(configlocation, "config.yml"), "r") as stream:
                 config_data = yaml.safe_load(stream)
                 print("Config loaded.\n")
+                param_checker(config_data)
                 return config_data
-
+                
     except FileNotFoundError:
         sys.exit("Directory containing config file, {}, not found. Exiting...".format(configlocation))
+        
+def param_checker(config_data):
+    """
+    Checks the parameters in the config file.
+
+    If finds a parameter that is incorrect, will prompt user what is incorrect.
+    Parameters
+    ----------
+    config_data : dict
+        Dictionary containing the data for parameters in the config file.
+
+    Returns
+    -------
+    None
+
+    """
+    #Obtain all the parameters in parsable way
+    average_number = config_data["advanced_parameters"]["average_number"]
+    conversion_factor = config_data["advanced_parameters"]["conversion_factor"]
+    cv_start_voltage = config_data["cyclic_voltammetry"]["start_voltage"]
+    cv_sweep_rate = config_data["cyclic_voltammetry"]["sweep_rate"]
+    cycle_number = config_data["cyclic_voltammetry"]["number_of_cycles"]
+    data_out_name = config_data["general_parameters"]["data_output_filename"]
+    data_out_path = config_data["general_parameters"]["data_output_path"]
+    end_voltage = config_data["linear_sweep_voltammetry"]["end_voltage"]
+    exp_time = config_data["chronoamperometry"]["time"]
+    exp_type = config_data["general_parameters"]["experiment_type"]
+    lsv_start_voltage = config_data["linear_sweep_voltammetry"]["start_voltage"]
+    first_turnover = config_data["cyclic_voltammetry"]["first_turnover_voltage"]
+    lsv_sweep_rate = config_data["linear_sweep_voltammetry"]["sweep_rate"]
+    rest_time = config_data["general_parameters"]["rest_time"]
+    second_turnover = config_data["cyclic_voltammetry"]["second_turnover_voltage"]
+    set_gain = config_data["advanced_parameters"]["setpoint_gain"]
+    set_offset = config_data["advanced_parameters"]["setpoint_offset"]
+    shunt_resistor = config_data["advanced_parameters"]["shunt_resistor"]
+    step_number = config_data["general_parameters"]["step_number"]
+    time_step = config_data["advanced_parameters"]["time_step"]
+    voltage = config_data["chronoamperometry"]["voltage"]
+    
+    #Check if every variable is of the correct type
+    for i in [data_out_name, data_out_path]:
+        if isinstance(i, str) == False:
+            stri = str(i)
+            sys.exit("Warning! \nThe value " + stri + " in config.yml is not a string. \nExiting...")
```

Comment:
better to use formatted strings here, or fstrings =>

```
"Warning! \nThe value {} in config.yml is not a string. \nExiting...".format(str(i))
```
and remove the ``stri = str(i)`` line.

Also, I think the error message is user facing so I would advocate for it to say something more meaningful to a non developer (Austin, I guess this is further evidence you are a developer now!).  Something like:

``"please check the {} entry in the config file, it doesn't seem to be a valid file-path".format(config_entry) ``  

would be much more helpful to the user....

Reply:
ah yeah I guess best way to point would be to point directly to entry label where the entry needs to be changed. Maybe where I assign the values I could make them into a dictionary where one entry is label in config while the other entry is the value?

Reply:
I changed the messages to be more precise and helpful by doing the following:

Changed initial dictionary parsing to load variables into dictionaries that contain the parameter name and the value that they return. This way, a message can be returned about what parameter is wrong and report back the incorrect value they provided. Additionally context and remedies were added to response messages.